src/components/Stats/StatsComponent.tsx

```diff
@@ -115,43 +125,109 @@ export class StatsComponent extends React.Component<WidgetProps> {
         this.height = height;
     };
 
+    onMouseEnter = () => {
+        this.showMouseEnterWidget();
+    };
+
+    onMouseLeave = () => {
+        this.hideMouseEnterWidget();
+    };
+
+    private getTableValue = (index, type) => {
+        let numString = "";
+        let unitString = "";
+        
+        if (isFinite(index) || type) {
```

Comment:
this OR logic is incorrect, should be if (this.statsData && isFinite(index) && index >= 0 && index < this.statsData.statistics?.length)